pkg/clients/github/pull_request.go

```diff
@@ -136,3 +136,36 @@ func (g *Github) GetCheckRunConclusion(checkRunName, repoName, prHeadSha string,
 	}
 	return checkRun.GetConclusion(), nil
 }
+
+// GetCheckRunStatus fetches a specific CheckRun within a given repo
+// by matching the CheckRun's name with the given checkRunName, and
+// then returns the CheckRun status
+func (g *Github) GetCheckRunStatus(checkRunName, repoName, prHeadSha string, prNumber int) (string, error) {
+	var errMsgSuffix = fmt.Sprintf("repository: %s, PR number: %d, PR head SHA: %s, checkRun name: %s\n", repoName, prNumber, prHeadSha, checkRunName)
+
+	var checkRun *github.CheckRun
+	var timeout time.Duration
+	var err error
+
+	timeout = time.Minute * 5
+
+	err = utils.WaitUntil(func() (done bool, err error) {
```

Comment:
what do you think if we add a timeout for the `WaitUntil` for avoiding endless loops

Reply:
Line 165 has a `timeout`

Reply:
you are right, 
my bad